crates/pop-contracts/src/call.rs

```diff
@@ -173,6 +176,53 @@ pub async fn call_smart_contract(
 	Ok(output)
 }
 
+/// Executes a smart contract call using a signed payload.
+///
+/// # Arguments
+///
+/// * `call_exec` - A struct containing the details of the contract call.
+/// * `payload` - The signed payload string to be submitted for executing the call.
+/// * `url` - The endpoint of the node where the call is executed.
+pub async fn call_smart_contract_from_signed_payload(
+	call_exec: CallExec<DefaultConfig, DefaultEnvironment, Keypair>,
+	payload: String,
+	url: &Url,
+) -> anyhow::Result<String, Error> {
+	let token_metadata = TokenMetadata::query::<DefaultConfig>(url).await?;
+	let metadata = call_exec.client().metadata();
+	let events = submit_signed_payload(url.as_str(), payload).await?;
+	let display_events = DisplayEvents::from_events::<SubstrateConfig, DefaultEnvironment>(
+		&events, None, &metadata,
+	)?;
+
+	let output =
+		display_events.display_events::<DefaultEnvironment>(Verbosity::Default, &token_metadata)?;
+	Ok(output)
+}
+
+/// Generates the payload for executing a smart contract call.
+///
+/// # Arguments
+/// * `call_exec` - A struct containing the details of the contract call.
+/// * `gas_limit` - The maximum amount of gas allocated for executing the contract call.
+pub fn get_call_payload(
+	call_exec: &CallExec<DefaultConfig, DefaultEnvironment, Keypair>,
+	gas_limit: Weight,
+) -> anyhow::Result<Vec<u8>> {
+	let storage_deposit_limit: Option<u128> = None;
```

Comment:
any reason not to use `call_exec.opts.storage_deposit_limit`?

Reply:
You are right, it makes more sense. Are you changing it in the `pop up contract` too? https://github.com/r0gue-io/pop-cli/pull/365/files#diff-1352e29eed34e9a7d6dadab47b3cba66c1838625bbc8533fdf9a5246bc8c1717R138

Reply:
https://github.com/r0gue-io/pop-cli/pull/378/commits/55367b00f420ca4461977eea16115961cd0c52c2